### PR TITLE
update max SciPy to 1.7.0

### PR DESCRIPTION
**Pull request by esc**
As title

### CODE REVIEW

**Comment by esc** [comment]
My guess is that the CI config doesn't activate, because the changes need to be on master?

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, will the conda recipes also need updating?
https://github.com/numba/numba-scipy/blob/9764341cf392090f08faa3052bc2a5d9b174e2d4/buildscripts/conda_recipes/numba-scipy/meta.yaml#L21-L30

**Comment by esc** [comment]
> Thanks for the patch, will the conda recipes also need updating?
> https://github.com/numba/numba-scipy/blob/9764341cf392090f08faa3052bc2a5d9b174e2d4/buildscripts/conda_recipes/numba-scipy/meta.yaml#L21-L30

fixed in ed96485dbf521a5a7b5a8eebf6f8299b2f71c1df

**Comment by esc** [comment]
Fixes #66

**Comment by de846** [comment]
Thanks for this PR 👍 Is it possible to get this merged anytime soon?

File: .github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        scipy-version: [">=0.16,<1.4", ">=1.4,<=1.6.2"]
+        scipy-version: [">=0.16,<1.4", ">=1.4,<=1.7.0"]
```
Comment (stuartarchibald) [inline .github/workflows/test_and_deploy.yml:21]: Think line 66 also needs the same change?

Comment (esc) [reply .github/workflows/test_and_deploy.yml:21]: fixed in e671fbb947

**Comment by esc** [comment]
@stuartarchibald I believe I have addressed everything in this pull-request. Let's get this merged and make a new PR if broken?

**Comment by esc** [comment]
@de846 thank you for the ping on this issue.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Comment by esc** [comment]
@stuartarchibald thank you for approval, will merge now and see what happens.

**Comment by esc** [comment]
So, after merging this, it seems like SciPy 1.7.0 isn't being picked up at all. Neither during build nor during  test:

During build I see:

```
Run conda install -c numba conda-build python=3.8 numba>=0.45 'scipy>=1.4,<=1.7.0' flake8 pytest pip
Obtaining file:///home/runner/work/numba-scipy/numba-scipy
Installing collected packages: numba-scipy
  Running setup.py develop for numba-scipy
Successfully installed numba-scipy
```

But this doesn't actually matter, because we do build with `conda-build` eventually. 

There we get:

```
Run conda build -c defaults -c numba --python 3.8 buildscripts/conda_recipes/numba-scipy/
...
    scipy:            1.6.2-py38had2a1c9_1     
```

Then during one of the test runs, I get:

```
Run conda install -c numba python=3.8 numba>=0.45 'scipy>=1.4,<=1.7.0' flake8 pytest
...
  scipy                     1.6.2            py38had2a1c9_1  
```

Probably it would make sense if someone else could confirm what I am seeing, to make sure I am not looking at the wrong output.

**Comment by stuartarchibald** [comment]
@esc is there a scipy 1.7.0 build in defaults?

**Comment by esc** [comment]
> @esc is there a scipy 1.7.0 build in defaults?

Anaconda.org does seem to imply this exists:

https://anaconda.org/anaconda/scipy

**Comment by esc** [comment]
I see, it's an off by one error, we should have used `1.7.1` not `1.7.0` apparently. I just tried making that change locally on my Mac and that got me the correct version.

**Comment by esc** [comment]
> I see, it's an off by one error, we should have used `1.7.1` not `1.7.0` apparently. I just tried making that change locally on my Mac and that got me the correct version.

Correct version yes, but I was unable to complete the package build:

```
Traceback (most recent call last):
  File "/Users/vhaenel/miniconda3/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/Users/vhaenel/miniconda3/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 474, in main
    execute(sys.argv[1:])
  File "/Users/vhaenel/miniconda3/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 465, in execute
    verify=args.verify, variants=args.variants)
  File "/Users/vhaenel/miniconda3/lib/python3.7/site-packages/conda_build/api.py", line 195, in build
    variants=variants
  File "/Users/vhaenel/miniconda3/lib/python3.7/site-packages/conda_build/build.py", line 3013, in build_tree
    notest=notest,
  File "/Users/vhaenel/miniconda3/lib/python3.7/site-packages/conda_build/build.py", line 2291, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/Users/vhaenel/miniconda3/lib/python3.7/site-packages/conda_build/build.py", line 1565, in bundle_conda
    "run that executable.".format(dep, env_var_name))
conda_build.exceptions.CondaBuildException: Empty package; python present in build and host deps.  You probably picked up the build environment's python  executable.  You need to alter your recipe to  use the PYTHON env var in your recipe to run that executable.
```

**Comment by esc** [comment]
A fix is here: https://github.com/numba/numba-scipy/pull/71

**Pull request merged**